implicits/src-3/upickle/implicits/Readers.scala

```diff
@@ -1,9 +1,117 @@
 package upickle.implicits
```

Comment:
This file is mostly just consolidated from `CaseClassReader.scala`; proper rewrite would be a separate PR